tests/src/test/scala/ironrust/FullIntegrationTest.scala

```diff
@@ -147,102 +176,13 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
       sdk = IronOxide.initialize(deviceContext, defaultConfig)
       val deviceList = sdk.userListDevices.getResult
 
-      newDeviceResult.getDevicePrivateKey.equals(newDeviceResult.getDevicePrivateKey) shouldBe true
-      deviceContext.equals(deviceContext) shouldBe true
       deviceList.length shouldBe 1
       deviceList.head.getId.getId shouldBe a[java.lang.Long]
       deviceList.head.getName.isPresent shouldBe true
       deviceList.head.getName.get shouldBe deviceName
     }
   }
 
-  "Class equality" should {
```

Comment:
So is the idea that we are checking that equals and hashcode are defined, and since we believe those implementations to be good because they are generated from a template, we don't need to check the result of equals and hashcode? We are relying on tests in ironoxide (or trusting derived implementations).

Assuming what I wrote is mostly correct, I can certainly follow that chain of logic although I'm not sure I'm 100% comfortable having no assertions at the ironoxide-java layer. In ironoxide if equality isn't defined we will get a compile error. If it's not defined here, we'll get the default implementation and equality (and hashcode) just won't _work_. 
Maybe I'm just being paranoid...

Reply:
The tests make sure that equals and hashCode are defined in the class itself, not inherited from `java.lang.Object`. rust_swig does not automatically implement the equals or hashCode functions, so it will only pass if we implemented them both ourselves, which we are now doing with the `pre_build_generate_equals_and_hashcode` macro. 
Personally, I think it's safe to trust the `derive` calls in ironoxide, and there are no places in ironoxide that we manually implement `PartialEq` or `Hash`, so as long as the implementations are good in the recrypt-rs layer, I don't see any problem with trusting them. 

>  If it's not defined here, we'll get the default implementation and equality (and hashcode) just won't work.

This is what will fail the tests, so unless we explicitly exclude the class, we will guarantee that we have implemented it, without even having remember to write tests for the new class.

All that to say, I think this change puts us in a far better place than we were before. It is more easily maintainable, it doesn't make any extra network calls, and it prevents us from forgetting to test classes in the future.

Reply:
> not inherited from java.lang.Object
I had missed that part. Nice! Carry on.